### PR TITLE
cf-promises/cf-promises.c: teach --show-vars to print JSON vars

### PR DESCRIPTION
**Pull request by tzz**
Simple change, picked from #2245 

@kacf should be an easy merge, docs and acceptance tests will come if the code is OK.

### CODE REVIEW

**Comment by kacf** [comment]
Regarding the changelog entry: We try to make them as user friendly as possible, including the mass of people that know nothing about the code. But it's still useful to include references to the source file like you do in your message, so in those cases it's better to make a separate entry, written from the user's perspective, like this:

```
cf-promises/cf-promises.c: teach --show-vars to print JSON vars

Changelog: cf-promises can now show JSON variables.
```

File: cf-promises/cf-promises.c [hunk 1]
```diff
@@ -474,15 +474,21 @@ static void ShowVariablesFormatted(EvalContext *ctx)
         char *varname = VarRefToString(v->ref, true);
 
         Writer *w = StringWriter();
-        RvalWrite(w, v->rval);
+        bool json = false;
```
Comment (kacf) [inline cf-promises/cf-promises.c]: This is unused now, AFAICS.

**Comment by kacf** [comment]
Looks good otherwise. I don't think we need explicit documentation for this, other than the changelog message, but an acceptance test would be nice! :-)

**Comment by tzz** [comment]
Ugh, I can't get the acceptance test to work easily without https://github.com/cfengine/core/pull/2460 so I hope it's OK to delay the test until https://github.com/cfengine/core/pull/2460 is merged (I'll put it on my TODO list) and merge this now.

**Comment by tzz** [comment]
Code adjusted as suggested.

**Comment by kacf** [comment]
#2460 should be quite quick to get in.

**Comment by kacf** [comment]
Ok, took a bit longer than I anticipated, but #2460 is in now.

**Comment by tzz** [comment]
OK, this is rebased now and with a full acceptance test, ready for final review.

**Comment by kacf** [comment]
Nice!

**Comment by kacf** [comment]
trigger build

**Pull request merged**